### PR TITLE
#55. Create Command Tests

### PR DESCRIPTION
**Pull request by mike-winberry**
Closes: #88 , #55 , #19, #20

### CODE REVIEW

File: src/cmd/generate/generate.go [hunk 1]
```diff
@@ -19,37 +20,41 @@ var GenerateCmd = &cobra.Command{
...
 }
 
-func GenerateCommand() *cobra.Command {
+func GenerateCommand(flags oscal.BaseFlags) (output []byte, err error) {
+	schemaBytes, err := os.ReadFile(flags.InputFile)
+	if err != nil {
+		return
+	}
+
+	tagList, err := utils.FormatTags(flags.Tags)
```
Comment (mike-winberry) [inline src/cmd/generate/generate.go:46]: #19

File: src/cmd/root.go [hunk 2]
```diff
@@ -1,34 +1,53 @@
...
-var rootCmd = &cobra.Command{
+var logFile string // -l, --log-file
+
+// RootCmd represents the base command when called without any subcommands
+var RootCmd = &cobra.Command{
 	Use:   "go-oscal",
 	Args:  cobra.ExactArgs(0),
 	Short: "Generate Go data types from OSCAL JSON schema",
 	Long:  "Generate Go data types from OSCAL JSON schema",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
```
Comment (mike-winberry) [inline src/cmd/root.go:23]: #20

File: src/internal/utils/files.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		err = CreateFileDirs(path)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create log file path: %s\n", err)
+		}
+		return OpenLogFile(path)
+	}
+	return file, nil
+}
+
+func CreateFileDirs(fullPath string) (err error) {
```
Comment (mike-winberry) [inline src/internal/utils/files.go:37]: #19

File: src/internal/utils/tags.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	"strings"
+)
+
+var validTags = map[string]bool{
+	"json": true,
+	"yaml": true,
+}
+
+// formatTags formats Go struct tags.
+func FormatTags(cmdlineTags string) (tagList []string, err error) {
```
Comment (mike-winberry) [inline src/internal/utils/tags.go:14]: #19

File: src/pkg/validation/utils.go [hunk 5]
```diff
@@ -108,9 +108,14 @@ func getModelType(model map[string]interface{}) (modelType string, err error) {
 // getOscalVersionFromMap returns formatted OSCAL version if valid version is passed, returns error if not.
 func getOscalVersionFromMap(model map[string]interface{}) (version string, err error) {
```
Comment (mike-winberry) [inline src/pkg/validation/utils.go:109]: #88

**Review by brandtkeller — APPROVED** [review]
Looks great - a few modifications we'll need to make downstream in Lula but otherwise the patterns your establishing for common structure looks good.

**Pull request merged**